solidity/test/integration/LlamaVesting.t.sol

```diff
@@ -17,47 +17,77 @@ contract IntegrationLlamaVesting is IntegrationBase {
     _vestingStartTime = _connextVestingWallet.start();
   }
 
-  function test_VestAndUnlock() public {
-    vm.prank(payer);
-    _llamaPay.depositAndCreate(TOTAL_AMOUNT, address(_connextVestingWallet), PAY_PER_SECOND);
-
+  function test_VestAndUnlock_LaunchDate() public {
     // Before the cliff
-    uint256 _timestamp = _connextVestingWallet.cliff() - 1;
-    uint256 _vestedAmount = (_timestamp - _vestingStartTime) * PAY_PER_SECOND / 1e2;
-
+    uint256 _timestamp = LAUNCH_DATE;
     // The unlocking contract holds the tokens
     _warpAndWithdraw(_timestamp);
-    _assertBalances(0);
-    assertEq(_nextToken.balanceOf(address(_connextVestingWallet)), _vestedAmount);
+    _assertWalletBalance(6_838_356 ether);
+    _assertOwnerBalance(0 ether);
+  }
+
+  function test_VestAndUnlock_1SecBeforeCliff() public {
+    // 1 secont before the cliff
+    uint256 _timestamp = LAUNCH_DATE + 365 days - 1;
+    _warpAndWithdraw(_timestamp);
+    _assertWalletBalance(13_078_355 ether);
+    _assertOwnerBalance(0 ether);
+  }
+
+  function test_VestAndUnlock_Cliff() public {
+    // Just at the cliff date
+    uint256 _timestamp = LAUNCH_DATE + 365 days;
+    _warpAndWithdraw(_timestamp);
+    _assertWalletBalance(11_158_356 ether);
+    _assertOwnerBalance(1_920_000 ether);
+  }
 
-    // After the 1st milestone
-    _warpAndWithdraw(_connextVestingWallet.cliff() + 10 days);
-    _assertBalances(2_551_232 ether);
+  function test_VestAndUnlock_1MonthAfterCliff() public {
+    // 1 month after the cliff
+    uint256 _timestamp = LAUNCH_DATE + 365 days + 365 days / 12;
+    _warpAndWithdraw(_timestamp);
+    _assertWalletBalance(9_758_356 ether);
+    _assertOwnerBalance(3_840_000 ether);
+  }
 
-    // Linear unlock after the 1st milestone
-    _warpAndWithdraw(_connextVestingWallet.cliff() + 365 days);
-    _assertBalances(12_480_118 ether);
+  function test_VestAndUnlock_1YearAfterCliff() public {
+    // 1 year after the cliff
+    uint256 _timestamp = LAUNCH_DATE + 365 days + 365 days;
+    _warpAndWithdraw(_timestamp);
+    _assertWalletBalance(0 ether);
+    _assertOwnerBalance(19_318_356 ether);
+  }
 
-    // After the unlocking period has ended
-    _warpAndWithdraw(_connextVestingWallet.cliff() + 365 days * 3 + 10 days);
-    _assertBalances(24_960_000 ether);
+  function test_VestAndUnlock_2YearsAfterCliff() public {
+    // 2 years after the cliff
+    uint256 _timestamp = LAUNCH_DATE + 365 days + 365 days + 365 days;
+    _warpAndWithdraw(_timestamp);
+    _assertWalletBalance(0 ether);
+    _assertOwnerBalance(24_960_000 ether);
```

Comment:
some comments and requests:
- [ ] i liked the previous all-in-one test, i think it was clearer, i'd prefer if you could unify them again 
- [ ] using `3 * YEAR` may be a good style
- [ ] blz rm the term `Cliff` from titles and comments, since doesn't make sense
- [ ] add `1/13s` in comments to explain the behaviour
- i VERY much like seeing the progress over the ys (and +yr/12) 👏 
- [ ] i think `LAUNCH_DATE + 3ys` doesn't precisely show when we'll get all tokens, i'd use `VEST_START_DATE + 4ys`